src/modules/dataset.ts

```diff
@@ -14,12 +14,20 @@ function updateDataset(oldVnode: VNode, vnode: VNode): void {
 
   for (key in oldDataset) {
     if (!dataset[key]) {
-      delete elm.dataset[key];
+      if (elm.dataset) {
+        delete elm.dataset[key];
+      } else {
+        elm.removeAttribute('data-' + key);
```

Comment:
keys used via `elm.dataset` are not reflective to the data attribute keys, they need transformation according to the spec. Here is an example:

```js
$0.setAttribute('data-foo-bar', 3);
// yield: DOMStringMap {fooBar: "3"}
$0.setAttribute('data-fooBar', 3);
// yield: DOMStringMap {fooBar: "3", foobar: "3"}
```

The proper transformation here is to apply a regexp to the key to make it a data key, e.g.:

```js
const CAPS_REGEX = /[A-Z]/g;
elm.removeAttribute('data-' + key.replace(CAPS_REGEX, (match: string): string => '-' + match.toLowerCase()));
```

something along those lines should be enough. yes, it will be a little bit slower for some old browsers, but the fast path is what really matters.